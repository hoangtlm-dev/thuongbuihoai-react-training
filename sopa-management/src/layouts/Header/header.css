.header {
  width: var(--w-extra-large);
  margin: 20px auto 0;
}

.header,
.header-menu,
.header-support {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-user {
  margin: 0 24px;
}
