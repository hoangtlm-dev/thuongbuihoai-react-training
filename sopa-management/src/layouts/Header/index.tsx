import {
  memo,
  useCallback,
  useEffect,
  useState
} from 'react';
import { shallow } from 'zustand/shallow';

// Components
import Image from '@components/common/Image';
import Menu,
{
  MenuTheme
} from '@components/common/Menu';
import Button,
{
  ButtonType
} from '@components/common/Button';
import Text,
{
  SizeType
} from '@components/common/Text';
import CartModal from '@components/CartModal';
import Logout from '@components/common/Logout';
import Icon, { IconType } from '@components/common/Icon';

// Constants
import { IMAGE } from '@constants/image';
import { MENU_HEADER } from '@constants/common';

// Hooks
import { useFetchCartProduct } from '@hooks/useQuery';

// Stores
import { useCartStore } from '@stores/cart';
import { useAuthenticationStore } from '@stores/login';

// Styles
import './header.css';

const Header = () => {
  const { data } = useFetchCartProduct();
  const [toggleModal, setToggleModal] = useState<boolean>(false);

  const [accounts] = useAuthenticationStore(
    (state) => [
      state.accounts,
    ],
    shallow
  );

  const imageHref = accounts.length === 0 ? '/login' : '/';

  const handleToggleModal = useCallback(() => {
    setToggleModal(!toggleModal);
  }, [toggleModal]);

  const [carts, setCarts] = useCartStore(
    (state) => [
      state.cart,
      state.setCart,
    ],
    shallow
  );

  useEffect(() => {
    if (data) {
      setCarts(data);
    }
  }, [data]);

  return (
    <>
      <div data-testid='header' className='header'>
        <Image
          href={imageHref}
          url={IMAGE.blackLogo}
          alt={IMAGE.blackLogo}
          width={144}
          height={24}
        />
        <div className='header-menu'>
          <Menu menuList={MENU_HEADER} type={MenuTheme.horizontal} />
          <Button
            ariaLabel='Shoe Finder Quiz'
            children='Shoe Finder Quiz'
            type={ButtonType.btnOutlinePrimary}
          />
        </div>
        <div className='header-support'>
          <Text
            text='Support'
            type={SizeType.normal}
            className='header-text' />
          <Logout className='header-user' />
          <Button
            ariaLabel='Shopping bag'
            children={<Icon iconName={IconType.bag} />}
            type={ButtonType.btnIconPrimary}
            onClick={handleToggleModal}
            className='header-btn'
          />
        </div>
      </div>
      {toggleModal &&
        <CartModal carts={carts} onToggleModal={handleToggleModal} />}
    </>
  );
};

export default memo(Header);
