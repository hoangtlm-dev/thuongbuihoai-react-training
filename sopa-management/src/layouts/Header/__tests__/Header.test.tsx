// Libs
import '@testing-library/jest-dom';

// Components
import Header from '@layouts/Header';

// Helpers
import { renderRouterTest } from '@helpers/testUtils';

describe('Header component', () => {
  test('should render Header component', () => {
    const { getByTestId } = renderRouterTest(<Header />);
    expect(getByTestId('header')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = renderRouterTest(<Header />);
    expect(container).toMatchSnapshot();
  });
});
