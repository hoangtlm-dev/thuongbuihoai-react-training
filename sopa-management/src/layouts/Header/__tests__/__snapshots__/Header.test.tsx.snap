// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <div
    class="header"
    data-testid="header"
  >
    <a
      data-testid="image-link"
      href="/"
    >
      <img
        alt="/images/logos/black-logo.svg"
        class=""
        src="/images/logos/black-logo.svg"
      />
    </a>
    <div
      class="header-menu"
    >
      <div
        class="menu"
        data-testid="menu"
      >
        <ul
          class="undefined menu-list-horizontal"
        >
          <p
            class="menu-main-horizontal"
          />
          <li
            class="undefined menu-item-horizontal"
          >
            <a
              href="/"
            >
              Men
            </a>
          </li>
          <li
            class="undefined menu-item-horizontal"
          >
            <a
              href="/"
            >
              Women
            </a>
          </li>
          <li
            class="undefined menu-item-horizontal"
          >
            <a
              href="/"
            >
              Accessories
            </a>
          </li>
          <li
            class="undefined menu-item-horizontal"
          >
            <a
              href="/"
            >
              About
            </a>
          </li>
        </ul>
      </div>
      <button
        class=" btn btn-out-line-primary"
        data-testid="button"
      >
        Shoe Finder Quiz
      </button>
    </div>
    <div
      class="header-support"
    >
      <p
        class="header-text text-normal"
        data-testid="text"
      >
        Support
      </p>
      <div
        class="header-user account"
        data-testid="logout"
      >
        <button
          class=" btn btn-icon-primary"
          data-testid="button"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 21V19C20 17.9391 19.5786 16.9217 18.8284 16.1716C18.0783 15.4214 17.0609 15 16 15H8C6.93913 15 5.92172 15.4214 5.17157 16.1716C4.42143 16.9217 4 17.9391 4 19V21"
              stroke="#02021D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M12 11C14.2091 11 16 9.20914 16 7C16 4.79086 14.2091 3 12 3C9.79086 3 8 4.79086 8 7C8 9.20914 9.79086 11 12 11Z"
              stroke="#02021D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
      <button
        class="header-btn btn btn-icon-primary"
        data-testid="button"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2L3 6V20C3 20.5304 3.21071 21.0391 3.58579 21.4142C3.96086 21.7893 4.46957 22 5 22H19C19.5304 22 20.0391 21.7893 20.4142 21.4142C20.7893 21.0391 21 20.5304 21 20V6L18 2H6Z"
            stroke="#02021D"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M3 6H21"
            stroke="#02021D"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M16 10C16 11.0609 15.5786 12.0783 14.8284 12.8284C14.0783 13.5786 13.0609 14 12 14C10.9391 14 9.92172 13.5786 9.17157 12.8284C8.42143 12.0783 8 11.0609 8 10"
            stroke="#02021D"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
